Test failure on master/circle - settings clients - sessions are listed in clients view

https://circleci.com/gh/mozilla/fxa-content-server/7471?utm_campaign=build-failed&utm_medium=email&utm_source=notification

> × firefox on any platform - settings clients - sessions are listed in clients view (5.063s)
AssertionError: expected 'Web Session,' to equal 'Web Session, node-XMLHttpRequest'
  at Function.assert.equal  <node_modules/intern/browser_modules/chai/chai.js:2277:10>
  at Command.<anonymous>  <tests/functional/lib/helpers.js:1582:16>
  at runCallback  <node_modules/leadfoot/Command.js:526:31>
  at Command.<anonymous>  <node_modules/leadfoot/Command.js:543:11>
  at <node_modules/dojo/Promise.ts:393:15>
  at run  <node_modules/dojo/Promise.ts:237:7>
  at <node_modules/dojo/nextTick.ts:44:3>
  at _combinedTickCallback  <internal/process/next_tick.js:73:7>
  at process._tickCallback  <internal/process/next_tick.js:104:9>
  at Command.then  <node_modules/leadfoot/Command.js:542:10>
  at Command.<anonymous>  <tests/functional/lib/helpers.js:1581:8>
  at Command.<anonymous>  <tests/functional/lib/helpers.js:55:25>
  at runCallback  <node_modules/leadfoot/Command.js:526:31>
  at Command.<anonymous>  <node_modules/leadfoot/Command.js:543:11>
  at <node_modules/dojo/Promise.ts:393:15>
  at run  <node_modules/dojo/Promise.ts:237:7>
  at <node_modules/dojo/nextTick.ts:44:3>
  at _combinedTickCallback  <internal/process/next_tick.js:73:7>
  at process._tickCallback  <internal/process/next_tick.js:104:9>
  at Command.then  <node_modules/leadfoot/Command.js:542:10>
  at Test.sessions are listed in clients view [as test]  <tests/functional/settings_clients.js:80:10>
  at <node_modules/intern/lib/Test.js:191:24>
  at <node_modules/intern/browser_modules/dojo/Promise.ts:393:15>
  at runCallbacks  <node_modules/intern/browser_modules/dojo/Promise.ts:11:11>
  at <node_modules/intern/browser_modules/dojo/Promise.ts:317:4>
  at run  <node_modules/intern/browser_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/browser_modules/dojo/nextTick.ts:44:3>
  at _combinedTickCallback  <internal/process/next_tick.js:73:7>
  at process._tickCallback  <internal/process/next_tick.js:104:9>

A quick screen cap of what shows in the content server:

<img width="509" alt="screen shot 2017-07-12 at 11 18 45" src="https://user-images.githubusercontent.com/848085/28113566-f6783224-66f4-11e7-971c-7dc0adc97828.png">

--- a/app/scripts/views/settings/clients.js
+++ b/app/scripts/views/settings/clients.js
@@ -59,8 +59,12 @@ define(function (require, exports, module) {
         if (item.lastAccessTimeFormatted) {
 
           if (item.isWebSession) {
-            item.title = this.translate(
-              t('Web Session, %(userAgent)s'), {userAgent: item.userAgent});
+            if (item.userAgent) {
+              item.title = this.translate(
+                t('Web Session, %(userAgent)s'), {userAgent: item.userAgent });
+            } else {
+              item.title = t('Web Session');
+            }
             item.lastAccessTimeFormatted = this.translate(
               t('%(translatedTimeAgo)s'), {translatedTimeAgo: item.lastAccessTimeFormatted});
           }

--- a/app/tests/spec/views/settings/clients.js
+++ b/app/tests/spec/views/settings/clients.js
@@ -515,11 +515,9 @@ define(function (require, exports, module) {
           });
       });
 
-      it('properly sets the title according to scope', () => {
+      it('properly sets the title', () => {
         return initView()
           .then(() => {
-            $('#container').html(view.el);
-
             const formatted = view._formatAccessTimeAndScope([
               {
                 clientType: 'oAuthApp',
@@ -543,6 +541,22 @@ define(function (require, exports, module) {
                 lastAccessTime: Date.now(),
                 lastAccessTimeFormatted: 'a month ago',
                 name: 'Add-ons'
+              },
+              {
+                deviceName: 'User\'s Web Session',
+                id: 'session-1',
+                isWebSession: true,
+                lastAccessTime: Date.now(),
+                lastAccessTimeFormatted: '12 minutes ago',
+                userAgent: 'Firefox 40'
+              },
+              {
+                deviceName: 'User\'s Second Web Session',
+                id: 'session-2',
+                isWebSession: true,
+                lastAccessTime: Date.now(),
+                lastAccessTimeFormatted: '18 minutes ago',
+                userAgent: ''
               }
             ]);
 
@@ -550,6 +564,8 @@ define(function (require, exports, module) {
             assert.equal(formatted[0].lastAccessTimeFormatted, 'last active a few seconds ago');
             assert.equal(formatted[1].title, 'Pocket - profile,profile:write');
             assert.equal(formatted[2].title, 'Add-ons');
+            assert.equal(formatted[3].title, 'Web Session, Firefox 40');
+            assert.equal(formatted[4].title, 'Web Session');
           });
       });
     });

--- a/tests/functional/settings_clients.js
+++ b/tests/functional/settings_clients.js
@@ -79,7 +79,7 @@ define([
         // second session is the node.js session from test setup
         .then(testElementTextEquals(
           '.client-webSession:nth-child(2) .client-name',
-          'Web Session, node-XMLHttpRequest'
+          'Web Session'
         ))
 
         // clicking disconnect on the second session should update the list

